### ISSUE TITLE
XF.4.5 with AndroidX Librarys crash in Release-Mode (ZXingScannerViewRenderer)

### ISSUE THREAD

**Issue by MickLesk** [Redth/ZXing.Net.Mobile#883]
Hi guys,

I have a project with your barcode scanner.
It works fine so far, but as soon as I change the release settings to "User+SDK assemblys", 
I can no longer build the app in the release configuration. 
The build process is aborted very quickly with the following exception.

```
System.IO.FileNotFoundException: Could not load assembly 'zxing.portable, Version=0.16.2.0, Culture=neutral, PublicKeyToken=830ae994e36ac27d'. Perhaps it doesn't exist in the Mono for Android profile?
Filename: "zxing.portable.dll"
   bei Java.Interop.Tools.Cecil.DirectoryAssemblyResolver.Resolve(AssemblyNameReference reference, ReaderParameters parameters) in /Users/runner/runners/2.165.0/work/1/s/xamarin-android/external/Java.Interop/src/Java.Interop.Tools.Cecil/Java.Interop.Tools.Cecil/DirectoryAssemblyResolver.cs:Zeile 241.
   bei Java.Interop.Tools.Cecil.DirectoryAssemblyResolver.Resolve(AssemblyNameReference reference) in /Users/runner/runners/2.165.0/work/1/s/xamarin-android/external/Java.Interop/src/Java.Interop.Tools.Cecil/Java.Interop.Tools.Cecil/DirectoryAssemblyResolver.cs:Zeile 191.
   bei Mono.Cecil.MetadataResolver.Resolve(TypeReference type) in /Users/builder/jenkins/workspace/archive-mono/2019-10/android/release/external/cecil/Mono.Cecil/MetadataResolver.cs:Zeile 110.
   bei Mono.Cecil.TypeReference.Resolve() in /Users/builder/jenkins/workspace/archive-mono/2019-10/android/release/external/cecil/Mono.Cecil/TypeReference.cs:Zeile 276.
   bei Mono.Linker.TypeReferenceExtensions.<GetMethods>d__4.MoveNext()
   bei Mono.Linker.Steps.TypeMapStep.TryMatchMethod(TypeReference type, MethodReference method)
   bei Mono.Linker.Steps.TypeMapStep.<GetBaseMethodsInInterfaceHierarchy>d__16.MoveNext()
   bei Mono.Linker.Steps.TypeMapStep.MapVirtualInterfaceMethod(MethodDefinition method)
   bei Mono.Linker.Steps.TypeMapStep.MapVirtualMethods(TypeDefinition type)
   bei Mono.Linker.Steps.TypeMapStep.MapType(TypeDefinition type)
   bei Mono.Linker.Steps.TypeMapStep.ProcessAssembly(AssemblyDefinition assembly)
   bei Mono.Linker.Steps.BaseStep.Process(LinkContext context)
   bei Mono.Linker.Pipeline.ProcessStep(LinkContext context, IStep step)
   bei Mono.Linker.Pipeline.Process(LinkContext context)
   bei MonoDroid.Tuner.Linker.Process(LinkerOptions options, ILogger logger, LinkContext& context)
   bei Xamarin.Android.Tasks.LinkAssemblies.Execute(DirectoryAssemblyResolver res)
   bei Xamarin.Android.Tasks.LinkAssemblies.RunTask()
   bei Xamarin.Android.Tasks.AndroidTask.Execute()	
```

**Comment by BartoszKopec** [comment]
I'm using nuget version 3.0.0-beta5 of ZXing.Net.Mobile and 4.6.0.800 of XF. The workaround for now is to switch Linking to "None" in Android Options in Properties of Xamarin.Android project. With this option the building and archiving to apk/aab is succeeded.

**Comment by standamikes** [comment]
try something like this, create a static class and static method which you're gonna call in OnCreate ..
```
public static class LinkerExceptions
  {
    public static void PleaseKeepThese()
    {
      var zxing= typeof(ZXing.Net.Mobile.Forms.Android.ZXingBarcodeImageViewRenderer).FullName;
    }
  }
```
You probably don't want to switch to None linking since it will make the final package too large. With this workaround you should be able to keep this settings on Sdk Assemblies Only

**Comment by BartoszKopec** [comment]
I can build with success on Release with this solution.
@standamikes I tried build Xamarin.Android project in Release and Linking Sdk Assemblies Only using your workaround but with no success.
But:

- I cloned this repo (`git clone https://github.com/Redth/ZXing.Net.Mobile --recurse-submodules`), 
- updated XF in it, 
- _removed `ValidateOnly = "$(_XFXamlCValidateOnly)"` from Xamarin.Forms.targed in 120 line (somehow this repo could not build with this line)_, 
- uninstalled all ZXing nugets from NetStandard and X.Android and referenced this local repo (I added it to my main solution as projects), 
- referenced zxing.dll to NetStandard and X.Android projects (this dll is located in ZXing.Net.Mobile folder)
- my OnCreate has (propably) all ZXing initialization references:
```c#
base.OnCreate(savedInstanceState);
...
Xamarin.Essentials.Platform.Init(Application);
ZXing.Net.Mobile.Forms.Android.Platform.Init();
MobileBarcodeScanner.Initialize(Application); //deprecated
LinkerExceptions.PleaseKeepThese();
...
Forms.Init(this, savedInstanceState);
LoadApplication(new App());
```

And now i can build X.Android in Release with linking Sdk Assemblies Only, which results in much smaller apk and bundle files.
